src/tools/rbd/action/Snap.cc

```diff
@@ -68,6 +70,10 @@ int do_list_snaps(librbd::Image& image, Formatter *f, bool all_snaps, librados::
       tt_str = tt_str.substr(0, tt_str.length() - 1);
     }
 
+    r = image.snap_is_protected(s->name.c_str(), &snap_protected);
```

Comment:
You can only use this method if the snapshot is in the ```RBD_SNAP_NAMESPACE_TYPE_USER``` namespace -- otherwise, for listing a group or trash snapshot, this will return ```-ENOENT``` and fail. Also, can you add an ```std::cerr``` line here to describe the failure should it happen?

See [1] for an example of a failed test.

[1] http://qa-proxy.ceph.com/teuthology/jdillaman-2018-09-06_21:00:38-rbd-wip-jd-testing-distro-basic-smithi/2986111/teuthology.log

Reply:
@dillaman thanks for your good advice, but I have an issue, if listing a group or trash snapshot, the snapshot's protected is true or false?

Reply:
@zy751713126 Those cannot get protected, so it would be false. It also might be nice if the table form of the display used "yes" when true and "" when false to match ```rbd ls -l```